src/microbe_stage/Microbe.Contact.cs

```diff
@@ -358,19 +382,36 @@ public void Damage(float amount, string source)
     /// <summary>
     ///   Returns true when this microbe can engulf the target
     /// </summary>
-    public bool CanEngulf(Microbe target)
+    public bool CanEngulf(IEngulfable target)
     {
+        if (target.PhagocytosisStep != PhagocytosisPhase.None)
+            return false;
+
+        // Can't engulf recently ejected objects, this act as a cooldown
+        if (expelledObjects.Any(m => m.Object == target))
```

Comment:
expelledObjects is a hash map, you should use the contains method for actually constant time check if a value is in there rather than loop with LINQ.

Reply:
I would but unfortunately CanEngulf takes in an IEngulfable and not the wrapper object.

Reply:
Oh yeah, you access `m.Object` here. I guess this is fine then. If you have extra energy you could consider if expelledObjects should even be a hash (and be just a normal list) if the lookups can't be performed using it, it doesn't have really a performance benefit at all, in fact the extra hashing it does may actually reduce performance in this usage role.

Reply:
Done that.